File: server/makc2022--nodejs--nestjs/src/layer4/domains/dummy-one-to-many/domain-resource.ts
/** Copyright (c) 2022 Maxim Kuzmin. All rights reserved. Licensed under the MIT License. */

/** Ресурс домена. */
export interface DomainResource {
  /** Получить имя запроса на получение элемента.
   * @returns Имя запроса.
   */
  getItemGetQueryName(): string;

  /** Получить имя запроса на получение списка.
   * @returns Имя запроса.
   */
  getListGetQueryName(): string;
}

File: server/makc2022--dotnet/Makc2022.Layer4.Sql.Domains.DummyMain/DomainResource.cs
﻿// Copyright (c) 2022 Maxim Kuzmin. All rights reserved. Licensed under the MIT License.

using Microsoft.Extensions.Localization;

namespace Makc2022.Layer4.Sql.Domains.DummyMain
{
    /// <summary>
    /// Ресурс домена.
    /// </summary>
    public class DomainResource : IDomainResource
    {
        #region Properties

        private IStringLocalizer<DomainResource> Localizer { get; }

        #endregion Properties

        #region Constructors

        /// <summary>
        /// Конструктор.
        /// </summary>
        /// <param name="localizer">Локализатор.</param>
        public DomainResource(IStringLocalizer<DomainResource> localizer)
        {
            Localizer = localizer;
        }

        #endregion Constructors

        #region Public methods

        /// <inheritdoc/>
        public string GetItemGetOperationName()
        {
            return Localizer["@@ItemGetOperationName"];
        }

        /// <inheritdoc/>
        public string GetListGetOperationName()
        {
            return Localizer["@@ListGetOperationName"];
        }

        #endregion Public methods
    }
}
